src/workload.rs

```diff
@@ -257,8 +259,26 @@ impl WorkloadManager {
         config: config::Config,
         metrics: Arc<Metrics>,
         awaiting_ready: readiness::BlockReady,
+        cert_manager: Box<dyn CertificateProvider>,
     ) -> anyhow::Result<WorkloadManager> {
-        let workloads: Arc<Mutex<WorkloadStore>> = Arc::new(Mutex::new(WorkloadStore::default()));
+        let (tx, mut rx) = watch::channel(Identity::default());
```

Comment:
`watch` is wrong here. This is for storing a single latest value. The current code will actually only trigger once with the last identity requested, instead of on each request. a mpsc is probably more appropriate (basically the same as Go channel)

Reply:
ah, fixed.